### PR TITLE
Proposed Improvements

### PR DESCRIPTION
**Pull request by drkno**
- More ES6
- Capitalisation
- Remove redundant code
- Sentences

### CODE REVIEW

File: kassy.json [hunk 1]
```diff
@@ -1,11 +1,11 @@
...
+    "name": "markov",
+    "version": 2.3,
+    "startup": "markov-module.js",
+    "command": "markov",
+    "help": [
+        [ "{{commandPrefix}}markov [corpus] [numWords]", "Generates a markov chain based on a corpus for a given number of words." ],
+        ["{{commandPrefix}}markov add [corpusName] [corpus]", "Adds a new corpus with name 'corpusName' and the text 'corpus'"],
+        ["{{commandPrefix}}markov list", "Lists the available corpora."]
+    ]
+}
```
Comment (fallaciousreasoning) [inline kassy.json:11]: Ah yeah, Mozilla has a two space indents rule. I'm going to have to change that :/ Was not a fan.

File: markov.js [hunk 2]
```diff
@@ -1,4 +1,36 @@
-function buildChains(corpus, chainLength) {
+const fixCapitalisation = sentences => {
+    for (let i = 0; i < sentences.length; i++) {
+        sentences[i][0] = sentences[i][0].charAt(0).toUpperCase() + sentences[i][0].slice(1);
+        for (let j = 0; j < sentences[i].length; j++) {
```
Comment (fallaciousreasoning) [inline markov.js:4]: I mean, if you really want. It seems kind of excessive. A better approach might be leaving apostrophes in when we make chains? We probably can't get away from capitalizing I/I'll if it's important to you but we're going to lose proper nouns anyway.

File: markov.js [hunk 3]
```diff
@@ -1,4 +1,36 @@
-function buildChains(corpus, chainLength) {
+const fixCapitalisation = sentences => {
+    for (let i = 0; i < sentences.length; i++) {
+        sentences[i][0] = sentences[i][0].charAt(0).toUpperCase() + sentences[i][0].slice(1);
```
Comment (fallaciousreasoning) [inline markov.js:3]: I'm pretty sure we could generate (theoretically) a one character sentence (a), which could break stuff. It's pretty unlikely but still

File: markov.js [hunk 4]
```diff
@@ -74,16 +106,24 @@ class MarkovGenerator {
     }
 
     generate(corpusName, length) {
-        if (!this.corpora[corpusName]) return "That is not a thing (no such corpus).";
+        if (!this.corpora[corpusName])
+            return 'That is not a thing (no such corpus).';
```
Comment (fallaciousreasoning) [inline markov.js:95]: If we're doing this should we add braces?

File: markov.js [hunk 5]
```diff
@@ -74,16 +106,24 @@ class MarkovGenerator {
...
 
-        let corpus = this.corpora[corpusName];
+        const corpus = this.corpora[corpusName];
+        const sentences = [];
```
Comment (fallaciousreasoning) [inline markov.js:98]: What does const even mean...? Is it just that we won't assign to the variable?

**Review by fallaciousreasoning — CHANGES_REQUESTED** [review]
Mostly LGTM but there were a couple of questions I've got

File: markov.js (hunk 3 above)
Comment (drkno) [reply markov.js:3]: It wouldn't have an issue, because you can index out of bounds in JS without error

File: markov.js (hunk 4 above)
Comment (drkno) [reply markov.js:95]: If you would like me to, I can.

File: markov.js (hunk 5 above)
Comment (drkno) [reply markov.js:98]: Assign once. An array is an object, so manipulation is only properties of the object (hence you aren't actually changing the object itself)

File: kassy.json (hunk 1 above)
Comment (drkno) [reply kassy.json:11]: So leave or fix?

File: markov.js (hunk 2 above)
Comment (drkno) [reply markov.js:4]: Its a search for "i" and "i'll" which was extended to all the others. I can remove them if you want?

File: markov.js (hunk 3 above)
Comment (fallaciousreasoning) [reply markov.js:3]: Of course you can.

File: kassy.json (hunk 1 above)
Comment (fallaciousreasoning) [reply kassy.json:11]: Yours is better.

File: markov.js (hunk 2 above)
Comment (fallaciousreasoning) [reply markov.js:4]: I think remove others, we'll make a better solution for quotes.

File: markov.js (hunk 4 above)
Comment (fallaciousreasoning) [reply markov.js:95]: Naaaaaaaah

File: markov.js (hunk 5 above)
Comment (fallaciousreasoning) [reply markov.js:98]: Sweet as

**Review by fallaciousreasoning — APPROVED** [review]

**Pull request merged**